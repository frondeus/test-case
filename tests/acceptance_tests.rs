#![cfg(test)]

mod acceptance {
    use itertools::Itertools;
    use std::env;
    use std::path::PathBuf;
    use std::process::Command;

    #[test]
    fn basic() {
        let output = Command::new("cargo")
            .current_dir(PathBuf::from("acceptance_tests").join("basic"))
            .args(&["test"])
            .output()
            .expect("cargo command failed to start");

        let lines = String::from_utf8_lossy(&output.stdout)
            .to_string()
            .lines()
            .sorted()
            .join("\n");
        insta::assert_display_snapshot!(lines);
    }

    #[test]
    fn hamcrest_assertions() {
        let output = Command::new("cargo")
            .current_dir(PathBuf::from("acceptance_tests").join("hamcrest_assertions"))
            .args(&["test"])
            .output()
            .expect("cargo command failed to start");

        let lines = String::from_utf8_lossy(&output.stdout)
            .to_string()
            .lines()
            .sorted()
            .join("\n");
        insta::assert_display_snapshot!(lines);
    }

    #[test]
    fn r#async() {
        let output = Command::new("cargo")
            .current_dir(PathBuf::from("acceptance_tests").join("async"))
            .args(&["test"])
            .output()
            .expect("cargo command failed to start");

        let lines = String::from_utf8_lossy(&output.stdout)
            .to_string()
            .lines()
            .sorted()
            .join("\n");
        insta::assert_display_snapshot!(lines);
    }
}
