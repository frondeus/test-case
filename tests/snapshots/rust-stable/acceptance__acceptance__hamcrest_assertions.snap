---
source: tests/acceptance_tests.rs
expression: lines

---
running 0 tests
running 10 tests
test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out
test result: ok. 10 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out
test test_cases::hamcrest_feature_works::can_be_followed_by_comment ... ok
test test_cases::hamcrest_feature_works::none_expects_it_is_none_ ... ok
test test_cases::hamcrest_feature_works::some_str_expects_it_is_all_some_not_none_ ... ok
test test_cases::hamcrest_feature_works::some_str_expects_it_is_has_str_ ... ok
test test_cases::hamcrest_feature_works::some_str_expects_it_is_some_ ... ok
test test_cases::removes_odd_numbers::_1_3_expects_it_is_empty_ ... ok
test test_cases::removes_odd_numbers::_2_3_expects_it_is_contains_2_ ... ok
test test_cases::removes_odd_numbers::_2_3_expects_it_is_len_1_ ... ok
test test_cases::removes_odd_numbers::_2_3_expects_it_is_not_contains_3_ ... ok
test test_cases::removes_odd_numbers::_2_4_expects_it_is_contains_vec_2_4_ ... ok
